src/app/clusters/icd-management-server/icd-management-server.cpp

```diff
@@ -433,10 +425,15 @@ bool emberAfIcdManagementClusterUnregisterClientCallback(CommandHandler * comman
 bool emberAfIcdManagementClusterStayActiveRequestCallback(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
                                                           const Commands::StayActiveRequest::DecodableType & commandData)
 {
-    ICDManagementServer server;
-    InteractionModel::Status status = server.StayActiveRequest(commandObj->GetAccessingFabricIndex());
-
-    commandObj->AddStatus(commandPath, status);
+#if CHIP_CONFIG_ENABLE_ICD_SERVER
+    IcdManagement::Commands::StayActiveResponse::Type response;
+    response.promisedActiveDuration = Server::GetInstance().GetICDManager().StayActiveRequest(commandData.stayActiveDuration);
+    commandObj->AddResponse(commandPath, response);
+#else
+    // Note: We only keep this line around for platforms examples that enable the ICD management server without building the sample
```

Comment:
But aren't those non-spec-compliant in that they claim the command in AcceptedCommandList?  I don't understand the setup that causes this code to be needed.

Reply:
It's for the CI. We need to put an `#if` around `response.promisedActiveDuration = Server::GetInstance().GetICDManager()` otherwise they will cause compilation error.

That being said we could just remove the `#else` part and have this comment to explain the `#if` part instead.

Reply:
From the results observed, the following CI will fail without an if `#i` `GetICDManager`:
Build example - ASR / ASR 
Build example - ESP32 / ESP32
Build example - Infineon / Infineon examples building
Build example - Linux ARM / Linux ARM Cross compile 
Build example - Mbed OS / Mbed OS examples building 
Build example - QPG / QPG 
Build example - TI CC26X2X7 / cc26x2x7  
Build example - TI CC32XX / cc32xx 
Tests / Test Suites - Darwin (no-ble-asan-clang) (pull_request)
Tests / REPL Tests - Linux (pull_request) 
Build example - ESP32 / ESP32_1 (pull_request)
...

From what I see, it seems that the lock app includes the icd management server cluster, without the actual command, but that the compiler still tries to build the emberAfcallbacks.

Reply:
This sounds like a build/layering/dependency issue. We should try to fix that instead.

Reply:
@bzbarsky-apple Understood, UnsupportedCOmmand was removed and everything in the callback is ifdefed.
@andy31415 An issue about the need for ifdefing was opened [here](https://github.com/project-chip/connectedhomeip/issues/32321)